snap/hooks/install

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash -e
+
+# get the values of $SNAP_DATA and $SNAP using the current symlink instead of
+# the default behavior which has the revision hard-coded, which breaks after
+# a refresh
+SNAP_DATA_CURRENT=${SNAP_DATA/%$SNAP_REVISION/current}
+SNAP_CURRENT=${SNAP/%$SNAP_REVISION/current}
+
+# install all the config files from $SNAP/config/SERVICE/res/configuration.toml
+# into $SNAP_DATA/config and replace $SNAP_COMMON environment variable in the config files
+mkdir -p "$SNAP_DATA/config"
+if [ ! -f "$SNAP_DATA/config/app-service-configurable/res/configuration.toml" ]; then
+    mkdir -p "$SNAP_DATA/config/res"
+    mkdir -p "$SNAP_DATA/config/res/http-export"
+    mkdir -p "$SNAP_DATA/config/res/mqtt-export"
+    mkdir -p "$SNAP_DATA/config/res/rules-engine"
+    mkdir -p "$SNAP_DATA/config/res/push-to-core"
+
+    for file in configuration.toml "http-export/configuration.toml" "mqtt-export/configuration.toml" "rules-engine/configuration.toml" "push-to-core/configuration.toml"; do
+        cp "$SNAP/config/res/$file" "$SNAP_DATA/config/res/$file"
```

Comment:
I think we should check if the file exists here too, the purpose of doing it this way is so that if a user/system integrator wants to pre-seed files into $SNAP_DATA when an image is built (or before a snap is installed), they can put config files here and the install hook will not copy the ones from the snap into the directory, but here the logic we have is that if the `res/configuration.toml` doesn't exist, we make the assumption none of the other http-export, mqtt-export, etc. files exist either which could be wrong.